overlord/configstate/configcore/netplan.go

```diff
@@ -92,18 +118,147 @@ func getNetplanFromSystem(key string) (result interface{}, err error) {
 	netplan := conn.Object("io.netplan.Netplan", "/io/netplan/Netplan")
 
 	if err := netplan.Call("io.netplan.Netplan.Config", 0).Store(&netplanConfigSnapshotBusAddr); err != nil {
-		if derr, ok := err.(dbus.Error); ok {
-			// Having no netplan config is *not* an error, we just
-			// do not support netplan config.
-			if isNoServiceOrMethodErr(derr) {
-				return nil, nil
-			}
+		return nil, err
+	}
+	logger.Debugf("using netplan config snapshot %v", netplanConfigSnapshotBusAddr)
+
+	netplanCfgSnapshot := conn.Object("io.netplan.Netplan", dbus.ObjectPath(netplanConfigSnapshotBusAddr))
+	return netplanCfgSnapshot, nil
+}
+
+func validateNetplanSettings(tr config.Conf) error {
+	// XXX: validate somehow once we support writing?
+	return nil
+}
+
+func isNetplanChange(chg string) bool {
+	return chg == "core.system.network.netplan" || strings.HasPrefix(chg, "core.system.network.netplan.")
+}
+
+func hasNetplanChanges(tr config.Conf) bool {
+	for _, chg := range tr.Changes() {
+		if isNetplanChange(chg) {
+			return true
+		}
+	}
+	return false
+}
+
+var storeReachableRetryWait = 1 * time.Second
+
+func handleNetplanConfiguration(tr config.Conf, opts *fsOnlyContext) error {
+	if !hasNetplanChanges(tr) {
+		return nil
+	}
+
+	var cfg map[string]interface{}
+	if err := tr.Get("core", "system.network.netplan", &cfg); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+
+	netplanCfgSnapshot, err := getNetplanCfgSnapshot()
+	// Having no netplan config is *not* an error, we just
+	// do not support netplan config.
+	if isNoServiceOrMethodErr(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// Use the default config that most Ubuntu Core have
+	originHint := "90-snapd-config"
```

Comment:
what's the relation with what is used by console-conf?

Reply:
I picked the name, we could use the same name that console-conf is using (00-snapd-config.yaml) but on systems that use cloud-init to boot the name is yet again different. netplan should be able to deal with multiple "origin" files, so it should not matter much but I'm happy of course to adjust as needed.

Reply:
you are correct. This name is fine except for one problem.  console-conf runs after seeding is done I think so at seeding if there's default for netplan conf we will write this,  which means that console-conf config itself later will have lower priority which is probably not what we want? I'm not entirely sure what the fix could be or if I'm misunderstand things.  Naively we could use something like use 0-snapd-config (< 00-snapd-config) if seeded is not set yet? is that too naive or not helpful?